plan/plan.go

```diff
@@ -40,7 +43,7 @@ type Plan struct {
 	// Populated after calling Calculate()
 	Changes *Changes
 	// DomainFilter matches DNS names
-	DomainFilter endpoint.DomainFilter
+	DomainFilter endpoint.DomainFilterInterface
```

Comment:
I thought about changing `endpoints.DomainFilter` to be an interface. The problem I face is that it is several providers like AlibabaCloud are relying on the internals ( mostly the Filters entry) I would like to keep the PR as small as possible even though it may lead to a need fo refactoring